website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+
+We also recommend that you set a minimum required version of the Terraform binary by setting the `required_version` in your `terraform` block. This ensures that other operators use a version of Terraform that has all of the features your configuration relies on.
+
+<CodeBlockConfig hideClipboard>
+
+```hcl
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "5.34.0"
+    }
+  }
+
+  required_version = ">= 1.7"
+}
+```
+
+</CodeBlockConfig>
+
+The above example pins the version of the `hashicorp/aws` provider to version `5.34.0`, and requires that operators use Terraform `1.7` or newer.
+
+For modules sourced from a registry, use the `version` parameter in the `module` block to pin the version. For local modules, Terraform ignores the `version` parameter.
+
+<CodeBlockConfig hideClipboard>
+
+```hcl
+module "vault_starter" {
+  source  = "hashicorp/vault-starter/aws"
+  version = "1.0.0"
+  ##...
+}
+```
+
+</CodeBlockConfig>
+
+## Module repository names
+
+The Terraform registry requires that repositories match a naming convention for all modules that you publish to the registry. Module repositories must use this three-part name `terraform-<PROVIDER>-<NAME>`, where `<NAME>` reflects the type of infrastructure the module manages and `<PROVIDER>` is the main provider the module uses. The `<NAME>` segment can contain additional hyphens, for example, `terraform-google-vault` or `terraform-aws-ec2-instance`.
+
+## Module structure
+
+Terraform modules allow you to create self-contained, reusable pieces of infrastructure-as-code. You can use modules to abstract away complex code and create multiple resources using a single declaration.
+
+Use modules to group together logically related resources. For example:
+
+- A networking module that defines a VPC, along with its subnets, gateway, and security groups.
+- An application module defining all resources required for each deployment. This stack could include web servers, databases, storage, and supported networking.
+
+Review the [module creation recommended pattern documentation](/terraform/tutorials/modules/pattern-module-creation) and [standard module structure](/terraform/language/modules/develop/structure) for guidance on how to structure your modules.
+
+## Local modules
+
+Local modules are sourced from local disk rather than a remote module registry. We recommend publishing your modules to a module registry, such as the [Terraform Cloud private module registry](/terraform/cloud-docs/registry). A module registry lets you share modules across your organization for use by other developers. If you cannot use a module registry, organizing part of your code in local modules still reduces the burden of maintaining and updating your code. 
```

Comment:
```suggestion
Local modules are sourced from local disk rather than a remote module registry. We recommend publishing your modules to a module registry, such as the [Terraform Cloud private module registry](/terraform/cloud-docs/registry), to easily version, share, and reuse modules across your organization. If you cannot use a module registry, using local modules can simplify maintaining and updating your code. 
```